release/charts/charts.go

```diff
@@ -0,0 +1,61 @@
+package charts
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	ecmConfig "github.com/rancher/ecm-distro-tools/cmd/release/config"
+)
+
+// ListLifecycleStatus prints the lifecycle status of the charts
+func ListLifecycleStatus(ctx context.Context, c *ecmConfig.ChartsRelease, branch, chart string) error {
+
+	var branchArg, chartArg string = "", ""
+
+	branchArg = "--branch-version=" + branch
+	if chart != "" {
+		chartArg = "--chart=" + chart
+	}
+
+	err := executeChartsBuildScripts(c.Workspace, "lifecycle-status", branchArg, chartArg)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("generated log files for inspection at: \n%s\n", c.Workspace+"/logs/")
+	return nil
+}
+
+func executeChartsBuildScripts(chartsRepoPath string, args ...string) error {
+	// save current working dir
+	ecmWorkDir, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	// change working dir to the charts repo
+	err = os.Chdir(chartsRepoPath)
+	if err != nil {
+		return err
+	}
+
+	bin := strings.Join([]string{chartsRepoPath, "bin", "charts-build-scripts"}, string(os.PathSeparator))
+
+	cmd := exec.Command(bin, args...)
+	output, err := cmd.CombinedOutput()
+	if err != nil {
```

Comment:
What is the contents of err if an error could occur?

Reply:
@briandowns: One small example:
```
Error: failed to execute charts-build-scripts: exit status 1, output: time="2024-07-26T18:33:23-03:00" level=info msg="Initializing dependencies for lifecycle-status"
time=2024-07-26T18:33:23-03:00 level=fatal msg=encountered error while initializing dependencies: no assets found in the repository

error: failed to execute charts-build-scripts: exit status 1, output: time="2024-07-26T18:33:23-03:00" level=info msg="Initializing dependencies for lifecycle-status"
time=2024-07-26T18:33:23-03:00 level=fatal msg=encountered error while initializing dependencies: no assets found in the repository
```

Reply:
It doesn't appear as though there needs to be additional context to these errors. What's returned from the call should suffice so no need to do additional wrapping.